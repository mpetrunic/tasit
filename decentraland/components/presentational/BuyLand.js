import React from "react";
import { StyleSheet, View, Text } from "react-native";
import { responsiveHeight } from "react-native-responsive-dimensions";
import PropTypes from "prop-types";
import LandForSale from "./LandForSale";
import Button from "./Button";
import Colors from "@constants/Colors";

export default function BuyLand(props) {
  const { landForSale, onBuy, accountInfo } = props;
  const {
    account,
    setupInProgress: waitingForAccountSetup,
    fundedWithEthers,
    fundedWithMana,
    approvedMarketplace,
  } = accountInfo;

  let waitingMessage = "";
  if (waitingForAccountSetup) {
    if (account == null) waitingMessage = "Generating account...";
    else if (!fundedWithEthers) waitingMessage = "Funding account with ETH...";
    else if (!fundedWithMana && !approvedMarketplace)
      waitingMessage = "Funding account with MANA and approving marketplace...";
    else if (!approvedMarketplace) waitingMessage = "Approving marketplace...";
    else if (!fundedWithMana) waitingMessage = "Funding account with MANA...";
  }

  return (
    <View style={styles.container}>
      <LandForSale landForSale={landForSale} />
      {waitingForAccountSetup ? (
        <React.Fragment>
          <View style={styles.buttonView}>
            <Button title="Buy" disabled={true} onPress={() => {}} />
          </View>
          <View style={styles.textRow}>
            <Text>{waitingMessage}</Text>
          </View>
        </React.Fragment>
      ) : (
        <View style={styles.buttonView}>
          <Button title="Buy" onPress={onBuy} />
        </View>
      )}
    </View>
  );
}

BuyLand.propTypes = {
  landForSale: PropTypes.object.isRequired,
  onBuy: PropTypes.func.isRequired,
  accountInfo: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  textRow: { flexDirection: "row" },
  container: {
    flex: 1,
    backgroundColor: Colors.backgroundColor,
    alignItems: "center",
    justifyContent: "center",
  },
  buttonView: {
    flexDirection: "row",
    marginTop: responsiveHeight(5),
  },
});
