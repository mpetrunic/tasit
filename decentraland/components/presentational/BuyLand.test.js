import React from "react";
import { shallow } from "enzyme";
import BuyLand from "./BuyLand";

describe("BuyLand", () => {
  it("renders the component", async () => {
    const landForSale = {
      id: 123,
      priceMana: 0,
      priceUSD: 0,
      asset: {
        id: -1,
        name: "Not found",
        img: "https://decentraland.org/images/logo-65f7b27caf.png",
      },
    };
    const onBuy = () => {};

    expect(
      shallow(<BuyLand landForSale={landForSale} onBuy={onBuy} />)
    ).toMatchSnapshot();
  });
});
