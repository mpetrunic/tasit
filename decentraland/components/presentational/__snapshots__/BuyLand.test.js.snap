// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyLand renders the component 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <LandForSale
    landForSale={
      Object {
        "asset": Object {
          "id": -1,
          "img": "https://decentraland.org/images/logo-65f7b27caf.png",
          "name": "Not found",
        },
        "id": 123,
        "priceMana": 0,
        "priceUSD": 0,
      }
    }
  />
  <Component
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 66.7,
      }
    }
  >
    <Button
      onPress={[Function]}
      title="Buy"
    />
  </Component>
</Component>
`;
